modules/card.js

```diff
@@ -1,92 +1,281 @@
 
     handleMessage(command, parameter, msg) {
         const cardName = parameter.toLowerCase();
-        return rp({
-            url: this.cardApi + cardName,
-            json: true
-        }).then(body => {
-            if (body.cards && body.cards.length) {
-                const card = this.findCard(cardName, body.cards);
-                let response = this.cardToString(card);
-                let otherCardNames = body.cards.filter(c => c.name !== card.name).map(c => "*" + c.name + "*");
-                if (otherCardNames.length) {
-                    otherCardNames.sort();
-                    otherCardNames = _.sortedUniq(otherCardNames);
-                    response += "\n\n:arrow_right: " + otherCardNames.length + " other matching cards: :large_blue_diamond:" + otherCardNames.join(" :large_blue_diamond:");
-                    if (response.length > this.maxLength) {
-                        response = response.substring(0, this.maxLength);
-                        response = response.substring(0, response.lastIndexOf(" :large_blue_diamond"));
-                        response += "\n\u2026";
+        // no card name, no lookup
+        if (!cardName) return;
+        // fetch data from API and Discord Guild
+        return Promise.all([
+            rp({url: this.cardApi + cardName, json: true}),
+            this.getEmojiPermission(msg)
+        ]).then(([body, permission]) => {
+            // check if there are results
+            if (body.data && body.data.length) {
+                // generate embed
+                this.generateEmbed(body.data, command, permission).then(embed => {
+                    return msg.channel.send('', {embed});
+                }).then(sentMessage => {
+                    // if multiple results, add reactions
+                    if (body.data.length > 1) {
+                        sentMessage.react('⬅').then(() => sentMessage.react('➡'));
+                        sentMessage.createReactionCollector(
+                            ({emoji} , user) => (emoji.toString() === '⬅' || emoji.toString() === '➡') && user.id === msg.author.id,
+                            {time: 30000, max: 20}
+                        ).on('collect', reaction => {
+                            //reaction.remove(msg.author); // needs edit message rights
+                            if(reaction.emoji.toString() === '⬅') {
+                                body.data.unshift(body.data.pop());
+                            } else {
+                                body.data.push(body.data.shift());
+                            }
+                            // edit the message to show the next card
+                            this.generateEmbed(body.data, command, permission).then(embed => {
+                                sentMessage.edit('', {embed});
+                            });
+                        }).on('end', () => {
+                            // clear body from memory
+                            body = null;
```

Comment:
Probably not necessary, and with Heroku restarting the bot every 24h it shouldn't leak too much in any case. I guess I might've been a bit paranoid there. Removed it.